client/src/components/LoginForm/SignupContent.jsx

```diff
@@ -2,61 +2,218 @@ import React, { Component } from 'react';
+    if (validation.isEmpty(password)) {
+      this.setError('Password is required.');
+      this.updateErrorState('passwordError', true);
+      isValid = false;
+    } else if (!validation.isPassword(password)) {
+      this.setError('Password must be between 8 and 64 characters.');
+      this.updateErrorState('passwordError', true);
+      isValid = false;
+    }
+    if (!validation.isEmail(email)) {
+      this.setError('Enter a valid email.');
+      this.updateErrorState('emailError', true);
+      isValid = false;
+    }
+    if (!validation.isStudentId(id)) {
+      this.setError('Enter a valid student ID.');
+      this.updateErrorState('usernameError', true);
+      isValid = false;
+    }
+    //TODO, add curiculum validation
+
+    return isValid;
+  }
+
+  displaySignupError(error) {
+    if (error === 'ER_DUP_ENTRY') {
+      this.setError('User with this student ID already exists.');
+    } else {
+      this.setError(error);
+    }
+  }
+
+  updateErrorState(field, isError) {
+    this.setState({
+      [field]: [isError],
+    });
+  }
+
+  setError(message) {
+    document.getElementById('error-msg').innerHTML = message;
+  }
+
+  handleInputChange(event = {}) {
+    const name = event.target && event.target.name;
+    const value = event.target && event.target.value;
+    this.setState({ [name]: value });
+  }
+
   render() {
     return (
       <Section className="form" flexDirection="column">
-        <Section className="subform" flexDirection="row">
-          <EditText label="Full Name" /> &nbsp;
-          <EditText label="Student ID" />
-        </Section>
+        <text className="error" id="error-msg" />
+        <EditText
+          required
+          label="Student ID"
+          id="id-input"
+          type="username"
+          name="username"
+          defaultValue={this.state.username}
+          onChange={this.handleInputChange}
+          error={this.state.usernameError}
+        />
         <EditText
           required
           label="Email"
-          id="outlined-email-input"
+          id="email-input"
           type="email"
+          name="email"
+          defaultValue={this.state.email}
+          onChange={this.handleInputChange}
+          error={this.state.emailError}
         />
         <Section className="subform" flexDirection="row">
           <EditText
             required
             label="Password"
-            id="outlined-password-input"
+            id="password-input"
             type="password"
+            name="password"
+            defaultValue={this.state.password}
+            onChange={this.handleInputChange}
+            error={this.state.passwordError}
           />
           &nbsp;
           <EditText
             required
             label="Confirm password"
-            id="outlined-password-input"
+            id="confirm-password-input"
             type="password"
+            name="confirmPassword"
+            defaultValue={this.state.confirmPassword}
```

Comment:
You're missing the confirmPassword state in your constructor, so you get a warning when the user tries to type in the Confirm password textbox.

> Warning: A component is changing an uncontrolled input of type password to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components